### PR TITLE
chore(react-tags-preview): update Tag stories with better a11y

### PR DESCRIPTION
**Pull request by YuanboXue-Amber**
1. Update Tag/InteractionTag dismiss story - instead of just a visual example, show dismiss usage

2. Update all stories using dismissible Tag/InteractionTag to have correct aria-label on the dismiss icon/button. Update bestPractices.md with this info.

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                |         Size (minified/GZIP) |
| ---------------------------------------------------------------------------------------------------------------- | ---------------------------: |
| <samp>react-tags-preview</samp> <br /> <abbr title='bundle-size/InteractionTag.fixture.js'>InteractionTag</abbr> |  `11.031 kB`<br />`4.528 kB` |
| <samp>react-tags-preview</samp> <br /> <abbr title='bundle-size/tag.fixture.js'>Tag</abbr>                       |  `25.407 kB`<br />`8.398 kB` |
| <samp>react-tags-preview</samp> <br /> <abbr title='bundle-size/TagGroup.fixture.js'>TagGroup</abbr>             | `69.607 kB`<br />`20.482 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/768de13c5e30222f64b2cc453bc4b134a6e90712'>768de13c5e30222f64b2cc453bc4b134a6e90712</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

File: packages/react-components/react-tags-preview/stories/InteractionTag/InteractionTagAppearance.stories.tsx [hunk 1]
```diff
@@ -19,19 +19,19 @@ export const Appearance = () => {
...
         </InteractionTagPrimary>
-        <InteractionTagSecondary />
+        <InteractionTagSecondary aria-label={`filled, remove`} />
       </InteractionTag>
       <InteractionTag appearance="outline">
         <InteractionTagPrimary icon={<CalendarMonth />} hasSecondaryAction>
           outline
         </InteractionTagPrimary>
-        <InteractionTagSecondary />
+        <InteractionTagSecondary aria-label={`outline, remove`} />
```
Comment (bsunderhus) [inline packages/react-components/react-tags-preview/stories/InteractionTag/InteractionTagAppearance.stories.tsx:28]: nit:
```suggestion
        <InteractionTagSecondary aria-label="outline, remove" />
```

**Review by bsunderhus — APPROVED** [review]

File: packages/react-components/react-tags-preview/stories/InteractionTag/InteractionTagAppearance.stories.tsx [hunk 2]
```diff
@@ -19,19 +19,19 @@ export const Appearance = () => {
         <InteractionTagPrimary icon={<CalendarMonth />} hasSecondaryAction>
```
Comment (fabricteam) [inline packages/react-components/react-tags-preview/stories/InteractionTag/InteractionTagAppearance.stories.tsx:19]: ## [🕵 fluentuiv9 Open the Visual Regressions report to inspect the affected screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=28909&commitId=31afd281a024d86332756a920eb906f9c20e7f96&env=prod&clientType=FLUENTUI)


  <details><summary><b>InteractionTag Converged</b> 6 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|InteractionTag Converged.Brand - Dark Mode.pressed dismiss.chromium.png|115| Changed|
|InteractionTag Converged.Brand.pressed dismiss.chromium.png|131| Changed|
|InteractionTag Converged.Filled - Dark Mode.pressed dismiss.chromium.png|119| Changed|
|InteractionTag Converged.Outline - Dark Mode.pressed dismiss.chromium.png|4| Changed|
|InteractionTag Converged.Filled.pressed dismiss.chromium.png|80| Changed|
|InteractionTag Converged.Outline.pressed dismiss.chromium.png|72| Changed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

**Pull request merged**